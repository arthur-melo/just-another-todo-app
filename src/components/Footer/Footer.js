import React from 'react';
import PropTypes from 'prop-types';

import './Footer.css';

const footerPropTypes = {
  className: PropTypes.string,
};

const Footer = ({ className }) => (
  <div className={className}>
    <div className="footer__container">
      <p className="footer__p">
        Made using React + Redux -{' '}
        <a href="https://github.com/arthur-melo" rel="noopener noreferrer" target="_blank">
          @arthur-melo
        </a>
      </p>
    </div>
  </div>
);

Footer.propTypes = footerPropTypes;

export default Footer;
