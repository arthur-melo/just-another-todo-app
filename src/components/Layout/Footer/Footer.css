.footer__container {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  background: var(--background-color);
}

.footer__p {
  margin: 0;
}
