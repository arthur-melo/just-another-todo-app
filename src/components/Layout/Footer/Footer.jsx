import { useState, useEffect } from 'react';

const Footer = () => {
  const [emoji, setEmoji] = useState();

  useEffect(() => {
    const emojiCombination = ['❤️', '💻', '✨', '🍕', '⚛️'];
    let counter = 0;

    setEmoji(emojiCombination.at(counter));
    counter++;

    const intervalId = setInterval(() => {
      if (counter >= emojiCombination.length) {
        counter = 0;
      }
      setEmoji(emojiCombination.at(counter));
      counter++;
    }, 5000);

    return () => clearInterval(intervalId);
  }, []);

  return (
    <div className="bg-light text-center d-flex h-100 justify-content-center align-items-center p-1">
      <div className="wrapper">
        {' '}
        <p className="w-100 m-0">
          This app runs offline and doesn't use analytics.
        </p>
        <p className="w-100 m-0">Your data is safe in your device!</p>
        <p className="w-100 m-0">
          Made with {emoji} by{' '}
          <a
            className="text-info"
            href="https://github.com/arthur-melo"
            rel="noopener noreferrer"
            target="_blank">
            @arthur-melo
          </a>
        </p>
      </div>
    </div>
  );
};

export default Footer;
