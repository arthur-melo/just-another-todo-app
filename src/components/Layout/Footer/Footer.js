import { useState, useEffect } from 'react';

import './Footer.css';

const Footer = () => {
  const [emoji, setEmoji] = useState();

  useEffect(() => {
    const emojiCombination = ['❤️', '💻', '✨', '🍕', '⚛️'];
    let counter = 0;

    setEmoji(emojiCombination.at(counter));
    counter++;

    const intervalId = setInterval(() => {
      if (counter >= emojiCombination.length) {
        counter = 0;
      }
      setEmoji(emojiCombination.at(counter));
      counter++;
    }, 5000);

    return () => clearInterval(intervalId);
  }, []);

  return (
    <div className="bg-light text-center d-flex h-100 justify-content-center align-items-center p-1">
      <div className="wrapper">
        {' '}
        <p className="footer__p">
          This app runs offline and doesn't use analytics.
        </p>
        <p className="footer__p">Keep your data safe in your browser!</p>
        <p className="footer__p">
          Made with {emoji} by{' '}
          <a
            className="text-info"
            href="https://github.com/arthur-melo"
            rel="noopener noreferrer"
            target="_blank">
            @arthur-melo
          </a>
        </p>
      </div>
    </div>
  );
};

export default Footer;
