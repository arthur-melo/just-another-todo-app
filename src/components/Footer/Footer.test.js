import React from 'react';

import { render } from '@testing-library/react';

import Footer from './Footer';

describe('Footer', () => {
  it('should render the Footer component', () => {
    const className = 'test-classname';

    const { container } = render(<Footer className={className} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
