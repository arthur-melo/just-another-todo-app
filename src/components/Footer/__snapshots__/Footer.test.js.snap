// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render the Footer component 1`] = `
<div
  class="test-classname"
>
  <div
    class="footer__container"
  >
    <p
      class="footer__p"
    >
      Made using React + Redux -
       
      <a
        href="https://github.com/arthur-melo"
        rel="noopener noreferrer"
        target="_blank"
      >
        @arthur-melo
      </a>
    </p>
  </div>
</div>
`;
